assets/js/base/components/product-list/style.scss

```diff
@@ -74,15 +74,25 @@
 			max-width: 100% / $i;
 		}
 	}
-	&.has-4-columns:not(.alignwide):not(.alignfull),
-	&.has-5-columns:not(.alignfull),
-	&.has-6-columns:not(.alignfull),
-	&.has-7-columns,
-	&.has-8-columns {
+	// Adjust font size as more cols are added.
+	&.has-6-columns .wc-block-grid__product {
+		font-size: 0.5em;
+	}
+	&.has-6-columns.alignfull,
+	&.has-5-columns {
 		.wc-block-grid__product {
-			font-size: 0.8em;
+			font-size: 0.58em;
 		}
 	}
+	&.has-5-columns.alignfull,
+	&.has-4-columns:not(.alignwide):not(.alignfull) {
+		.wc-block-grid__product {
+			font-size: 0.75em;
+		}
+	}
+	&.has-3-columns:not(.alignwide):not(.alignfull) {
+		font-size: 0.92em;
```

Comment:
I didn't test it, but do you think `0.875em` would work here? In that case we could use the [`font-size()` mixin](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/css/abstracts/_mixins.scss#L9-L12) with `small` size. And `0.75em` could be replaced with `font-size("smaller")`.

For `0.5em` and `0.58em`, we can either create new named font sizes in the mixin, or leave the hard-coded values here. I don't have a strong opinion.